stl/inc/atomic

```diff
@@ -351,6 +373,33 @@ struct _Atomic_storage {
         return _Result;
     }
 
+#if _HAS_CXX20
+
+    void wait(_Ty _Expected, const memory_order _Order = memory_order_seq_cst) const noexcept {
+        long _Spin_context = 0;
+        for (;;) {
+            _Ty _Observed = load(_Order);
+            if (_CSTD memcmp(&_Observed, _STD addressof(_Expected), sizeof(_Ty)) != 0) {
+                break;
+            }
+            __std_atomic_wait_indirect(_STD addressof(_Storage), _Spin_context);
+        }
+        if (_Spin_context & _ATOMIC_UNWAIT_NEEDED)
+            [[unlikely]] {

```

Comment:
`[[unlikely]]` needs to be macroized because it emits warnings (and has no effect) in earlier-than-C++20 modes. Also, if you didn't actually observe a codegen difference we shouldn't put it here...